src/App.vue

```diff
@@ -1,29 +1,54 @@
 <template>
-  <div id="app">
-    <img
-      alt="Vue logo"
-      src="./assets/logo.png"
-    />
-    <HelloWorld msg="Hello Vue 2 + Vite" />
+  <div>
+    <the-header></the-header>
```

Comment:
Nitpick: It's common to prefix components with at least a single letter of domain or company you are working on.  
In your case you could name your imports like `import VHeader` from './components/layout/Header.vue';
and then use it `<v-header>`.
But, as I mentioned, it's just a nitpick.

Reply:
I just did what was recommended in style guide https://vuejs.org/v2/style-guide/#Single-instance-component-names-strongly-recommended 
But probably there is few solutions.

Reply:
Roger that! ;)